apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -208,18 +227,17 @@ class SectionRow extends Component {
 
   render() {
     const {
-      section,
+      sections,
+      sectionId,
       validLoginTypes,
       validGrades,
       validAssignments
     } = this.props;
     const { editing, deleting } = this.state;
 
-    // When deleting a section, I've occasionally seen us attempt a render with
-    // a null section for some reason. This is here to provide defense against this.
-    if (!section) {
-      return null;
-    }
+    const section = sections[sectionId];
```

Comment:
Does this solve the problem?  You're still getting `sections` from redux and `sectionId` from the parent component.  Seems like the ordering problem you ran into before could still happen?